cmake/OpenCVFindLAPACK.cmake

```diff
@@ -106,11 +106,37 @@ macro(ocv_lapack_check)
       list(APPEND __link_directories ${LAPACK_LINK_LIBRARIES})
     endif()
 
+    set(LAPACK_TRY_COMPILE_DEF "")
+    if(LAPACK_IMPL STREQUAL "LAPACK/Apple") # https://github.com/opencv/opencv/issues/24660
+      # TODO: Need to find a way detecting macOS / iOS versions
+      #       enbale if macOS >= 13.3 or iOS >= 16.4
+
+      # Get macOS version
+      execute_process(COMMAND sw_vers -productVersion
+                      OUTPUT_VARIABLE MACOS_VERSION
+                      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+      # Extract major and minor versions
+      string(REGEX MATCHALL "[0-9]+" MACOS_VERSION_NUMBER_LIST ${MACOS_VERSION})
+      list(LENGTH MACOS_VERSION_NUMBER_LIST MACOS_VERSION_NUMBER_LIST_LENGTH)
+      if(MACOS_VERSION_NUMBER_LIST_LENGTH GREATER_EQUAL 2)
+        list(GET MACOS_VERSION_NUMBER_LIST 0 MACOS_MAJOR_VERSION)
+        list(GET MACOS_VERSION_NUMBER_LIST 1 MACOS_MINOR_VERSION)
+      endif()
+      # Enable Accelerate New LAPACK if macOS >= 13.3
+      if(MACOS_MAJOR_VERSION GREATER_EQUAL 13 AND MACOS_MINOR_VERSION GREATER_EQUAL 3)
```

Comment:
It does not work for 14.2, for example. Please use special version comparison: https://cmake.org/cmake/help/latest/command/if.html#version-comparisons. Example: https://github.com/opencv/opencv/blob/f9dd20eb07210f4759889f667fc8fc1ad90f55da/CMakeLists.txt#L865

Reply:
Done.